python/cppyy/numba_ext.py

```diff
@@ -130,14 +142,59 @@ def get_call_type(self, context, args, kwds):
         except KeyError:
             pass
 
-        ol = CppFunctionNumbaType(self._func.__overload__(*(numba2cpp(x) for x in args)), self._is_method)
+        args_mapping = []
+        for i, arg in enumerate(list(args)):
+            match = re.search(r"typeref\[(.*?)\*\]", str(arg))
+            if match:
+                literal_val = match.group(1)
+                arg_type = numba.typeof(eval(literal_val))
+                args_mapping.append(to_ref(numba2cpp(arg_type)))
+            else:
+                args_mapping.append(numba2cpp(arg))
+        args_combinations = itertools.product(*args_mapping)
+
+        for arg_combo in args_combinations:
+            try:
+                if(len(arg_combo) > 1):
+                    signature = ", ".join(arg_combo)
+                elif(len(arg_combo) == 0):
+                    signature = ""
+                else:
+                    signature = arg_combo[0]
+                ol = CppFunctionNumbaType(self._func.__overload__(signature), self._is_method)
+                self._arg_set_matched = arg_combo
+                break
+            except:
+                # TODO : Find a better way to handle this error case. Despite a succesful __overload__ match,
+                #  the error could arise from failures in the lowering pass
+                # print("__overload__ call failed")
+                pass
 
         thistype = None
         if self._is_method:
             thistype = nb_types.voidptr
 
+        return_type = ol._func.__cpp_reflex__(cpp_refl.RETURN_TYPE)
+        is_reference_return = False
+
+        # TODO : This works for builtin reference type returns but not for objects, for now leaving it commented out
+        #  assuming no reference type tests are added. Use typeof_scope() and eventually merge this logic into cpp2numba
+        # if not str(type(return_type)).startswith("<class"):
+        #     if return_type.endswith("&"):
+        #         is_reference_return = True
+        #         return_type = return_type[:-1]
+        #     else:
+        #         is_reference_return = False
+
+        # print("Is reference:", is_reference_return)
```

Comment:
It's better to leave this out and put this in another branch if it is something you are working on.